### PR TITLE
odb: support new lef58 mincut, widthtable, arrayspacing

### PR DESCRIPTION
**Pull request by osamahammad21**
closes [OR-1391](https://openroad.atlassian.net/browse/OR-1391)

### CODE REVIEW

File: src/odb/src/db/dbModule.cpp [hunk 1]
```diff
@@ -175,16 +176,6 @@ const char* dbModule::getName() const
   return obj->_name;
 }
 
-std::string dbModule::getHierarchicalName() const
```
Comment (maliberty) [inline src/odb/src/db/dbModule.cpp:178]: Thanks for catching and fixing this.

File: src/odb/src/db/dbTechLayerMinCutRule.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2020, The Regents of the University of California
```
Comment (maliberty) [inline src/odb/src/db/dbTechLayerMinCutRule.h:4]: Time to update the template year

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
You will need to update db_schema_initial & db_schema_minor in dbDatabase.h as we are now using .odb files and have to prevent cross loading across schema changes.

**Review by maliberty — CHANGES_REQUESTED** [review]
schema id

**Comment by osamahammad21** [comment]
> You will need to update db_schema_initial & db_schema_minor in dbDatabase.h as we are now using .odb files and have to prevent cross loading across schema changes.

@maliberty I am not familiar with those schema id. Can you explain what each one refers to?

**Comment by maliberty** [comment]
They are the major/minor version numbers for the internal schema.  If you make a change the will result in a change to the .odb file you need to update the schema so that it can be detected.  The 'initial' schema seems to be for backwards compatibility where a version X schema could load a X-1 .odb by having conditional paths in the loading.  We aren't using that for now so simply updating them is enough to prevent version errors.

**Review by maliberty — APPROVED** [review]
@arlpetergadfort FYI

**Pull request merged**